File: app/src/main/java/tpcreative/co/qrscanner/ui/create/GenerateCell.java
package tpcreative.co.qrscanner.ui.create;

/**
 * Created by Oclemy on 2017 for ProgrammingWizards TV Channel and http://www.camposha.info.
 - Our galaxycell class
 */

import android.content.Context;
import android.graphics.PorterDuff;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.appcompat.widget.AppCompatTextView;

import com.jaychang.srv.SimpleCell;
import com.jaychang.srv.SimpleViewHolder;
import butterknife.BindView;
import butterknife.ButterKnife;
import tpcreative.co.qrscanner.R;
import tpcreative.co.qrscanner.model.QRCodeType;


public class GenerateCell extends SimpleCell<QRCodeType,GenerateCell.ViewHolder> {
    private ItemSelectedListener listener;
    private static final String TAG = GenerateCell.class.getSimpleName();

    public GenerateCell(@NonNull QRCodeType item) {
        super(item);
    }

    protected void setListener(ItemSelectedListener listener){
        this.listener = listener;
    }

    @Override
    protected int getLayoutRes() {
        return R.layout.generate_item;
    }

    /*
    - Return a ViewHolder instance
     */
    @NonNull
    @Override
    protected ViewHolder onCreateViewHolder(ViewGroup parent, View cellView) {
        return new ViewHolder(cellView);
    }

    /*
    - Bind data to widgets in our viewholder.
     */
    @Override
    protected void onBindViewHolder(@NonNull ViewHolder viewHolder, final int i, @NonNull Context context, Object o) {
        final QRCodeType data = getItem();
        viewHolder.tvName.setText(data.name);
        viewHolder.imgIcon.setImageDrawable(context.getResources().getDrawable(data.res));
        viewHolder.imgDefault.setImageDrawable(context.getResources().getDrawable(R.drawable.baseline_add_box_white_48));
        viewHolder.imgDefault.setColorFilter(context.getResources().getColor(R.color.colorAccent), PorterDuff.Mode.SRC_ATOP);
        viewHolder.llRoot.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (listener!=null){
                    listener.onClickItem(i,false);
                }
            }
        });
    }
    /**
     - Our ViewHolder class.
     - Inner static class.
     * Define your view holder, which must extend SimpleViewHolder.
     * */
    static class ViewHolder extends SimpleViewHolder {
        @BindView(R.id.tvName)
        AppCompatTextView tvName;
        @BindView(R.id.imgIcon)
        AppCompatImageView imgIcon;
        @BindView(R.id.imgDefault)
        AppCompatImageView imgDefault;
        @BindView(R.id.llRoot)
        LinearLayout llRoot;
        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

    public  interface ItemSelectedListener{
        void onClickItem(int position, boolean isChecked);
        void onClickShare(String value);
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/ui/create/GenerateCell.kt
package tpcreative.co.qrscanner.ui.create
import android.content.Context
import android.graphics.PorterDuff
import android.view.*
import android.widget.LinearLayout
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.content.ContextCompat
import com.jaychang.srv.SimpleCell
import com.jaychang.srv.SimpleViewHolder
import kotlinx.android.synthetic.main.generate_item.view.*
import tpcreative.co.qrscanner.R
import tpcreative.co.qrscanner.model.QRCodeType

class GenerateCell(item: QRCodeType) : SimpleCell<QRCodeType, GenerateCell.ViewHolder>(item) {
    private var listener: ItemSelectedListener? = null
    fun setListener(listener: ItemSelectedListener?) {
        this.listener = listener
    }

    override fun getLayoutRes(): Int {
        return R.layout.generate_item
    }

    /*
    - Return a ViewHolder instance
     */
    protected override fun onCreateViewHolder(parent: ViewGroup, cellView: View): ViewHolder {
        return ViewHolder(cellView)
    }

    /*
    - Bind data to widgets in our viewholder.
     */
    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int, context: Context, o: Any?) {
        val data = item
        viewHolder.tvName.text = data.name
        viewHolder.imgIcon.setImageDrawable(ContextCompat.getDrawable(context,data.res))
        viewHolder.imgDefault.setImageDrawable(ContextCompat.getDrawable(context,R.drawable.baseline_add_box_white_48))
        viewHolder.imgDefault.setColorFilter(ContextCompat.getColor(context,R.color.colorAccent), PorterDuff.Mode.SRC_ATOP)
        viewHolder.llRoot.setOnClickListener(View.OnClickListener {
            if (listener != null) {
                listener?.onClickItem(i, false)
            }
        })
    }

    class ViewHolder(itemView: View) : SimpleViewHolder(itemView) {
        val tvName: AppCompatTextView = itemView.tvName
        val imgIcon: AppCompatImageView = itemView.imgIcon
        val imgDefault: AppCompatImageView = itemView.imgDefault
        val llRoot: LinearLayout = itemView.llRoot
    }

    interface ItemSelectedListener {
        fun onClickItem(position: Int, isChecked: Boolean)
        fun onClickShare(value: String?)
    }

    companion object {
        private val TAG = GenerateCell::class.java.simpleName
    }
}